core/transfer/src/main/java/org/eclipse/dataspaceconnector/transfer/core/transfer/TransferProcessManagerImpl.java

```diff
@@ -145,6 +151,183 @@ public void enqueueCommand(TransferProcessCommand command) {
         commandQueue.enqueue(command);
     }
 
+    /**
+     * Process INITIAL transfer<br/>
+     * set it to PROVISIONING
+     *
+     * @param process the INITIAL transfer fetched
+     * @return if the transfer has been processed or not
+     */
+    @WithSpan
+    private boolean processInitial(TransferProcess process) {
+        var manifest = manifestGenerator.generateResourceManifest(process);
+        process.transitionProvisioning(manifest);
+        transferProcessStore.update(process);
+        observable.invokeForEach(l -> l.provisioning(process));
+        return true;
+    }
+
+    /**
+     * Process PROVISIONING transfer<br/>
+     * Launch provision process. On completion, set to PROVISIONED if succeeded, ERROR otherwise
+     * <br/>
+     * On a consumer, provisioning may entail setting up a data destination and supporting infrastructure. On a provider, provisioning is initiated when a request is received and
+     * map involve preprocessing data or other operations.
+     *
+     * @param process the PROVISIONING transfer fetched
+     * @return if the transfer has been processed or not
+     */
+    @WithSpan
+    private boolean processProvisioning(TransferProcess process) {
+        provisionManager.provision(process)
+                .whenComplete((responses, throwable) -> {
+                    if (throwable == null) {
+                        onProvisionComplete(process.getId(), responses);
+                    } else {
+                        transitionToError(process.getId(), throwable, "Error during provisioning");
+                    }
+                });
+
+        return true;
+    }
+
+    /**
+     * Process PROVISIONED transfer<br/>
+     * If CONSUMER, set it to REQUESTING, if PROVIDER initiate data transfer
+     *
+     * @param process the PROVISIONED transfer fetched
+     * @return if the transfer has been processed or not
+     */
+    @WithSpan
+    private boolean processProvisioned(TransferProcess process) {
+        if (CONSUMER == process.getType()) {
+            process.transitionRequesting();
```

Comment:
nit: How about calling sendConsumerRequest here? This state seems to have only the effect of "wasting" a cycle. See https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/issues/336#issuecomment-1060026575. This can also be done in a subsequent PR for that issue, so we keep the scope of this PR.

Reply:
This was done to keep the pattern of having a `*-ING` and `*-ED` state, and this helps also setting up a clearer state separation.
e.g. at the moment there's no retry pattern applied on this "request", but, when we'll introduce it, on a failure the state of the transfer process will be set to `REQUESTING` to be retried again. Without this state, we would have to set it to `PROVISIONED`, but it will become unclear and could lead to errors in the case that new states pop up between `PROVISIONED` and `REQUESTED`.
It's like a step, step before we were talking about provision, in this step we're talking about request to the provider.